### PR TITLE
Eliminate Task.task_all warning

### LINKED ISSUE #8211: Eliminate the `Task.all_tasks()` warning in the CI Tests
**Python version**:
Python 3.6
Python 3.7
Python 3.8

**Operating system** (windows, osx, ...):
Ubuntu
Windows

**Issue**:
This Issue is part of the effort to close the [Issue#7738](https://github.com/RasaHQ/rasa/issues/7738) hence to reduce the amount of warnings that are raised when running all tests in the CI. From a first investigation, it seems that this warning (_PendingDeprecationWarning: Task.all_tasks() is deprecated, use asyncio.all_tasks() instead..._) for which the current Issue was opened, had 46 occurrences but in a latest CI run it appears in only ~27 tests.

**Error (including full traceback)**:
```
2021-03-15T14:21:47.6645370Z tests/test_train.py: 27 warnings
2021-03-15T14:21:47.6646399Z   /home/runner/work/rasa/rasa/rasa/utils/common.py:310: PendingDeprecationWarning: Task.all_tasks() is deprecated, use asyncio.all_tasks() instead
2021-03-15T14:21:47.6647500Z     pending = asyncio.Task.all_tasks()
```

**Possible Solutions**:
- As indicated in the Deprecation message, first step it would be to replace the `Task.all_tasks()` call with `asyncio.all_tasks()`.
- This however introduces another error 
```
E           RuntimeError: no running event loop
```
which would need to be addressed. 
- Related [docs](https://docs.python.org/3/library/asyncio-task.html#asyncio.all_tasks) 

**Command or request that led to error**:
- The warning can be found in any of the following CI tests log files :
  - 32_Run Tests (test-full-model-training, ubuntu-latest, 3.7).txt
  - 33_Run Tests (test-full-model-training, ubuntu-latest, 3.8).txt
  - 35_Run Tests (test-full-model-training, windows-latest, 3.7).txt
- Also, the warning can be reproduced when running one of the tests below:
```
pytest tests/test_train.py::TestE2e::test_e2e_gives_experimental_warning
pytest tests/test_telemetry.py::test_events_schema
```

**Definition of Done**:
- The `PendingDeprecationWarning: Task.all_tasks()` does not appear anymore.

### PR DESCRIPTION
**Pull request by iurisevero**
Resolve #8211 
**Proposed changes**:
- Update asyncio.Task.all_tasks() to asyncio.all_tasks
- Add a try catch on asyncio.all_tasks() to avoid error on python 3.6

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @rctatman will take a look at it as soon as possible ✨

**Comment by HotThoughts** [comment]
👋 Hey, the failing step `wait for docs test` in the GitHub Action is fixed. You should be able to pass it after updating your branch with the base branch.

File: rasa/utils/common.py [hunk 1]
```diff
@@ -309,7 +309,10 @@ def run_in_loop(
     result = loop.run_until_complete(f)
 
     # Let's also finish all running tasks:
-    pending = asyncio.Task.all_tasks()
+    try:
+        pending = asyncio.all_tasks(loop)
+    except AttributeError:
+        pending = asyncio.Task.all_tasks()
```
Comment (wochinge) [inline rasa/utils/common.py:315]: This is a fallback for python 3.6, right? Would be great to add a comment about it

**Review by wochinge — COMMENTED** [review]
Thanks for your help!

File: rasa/utils/common.py (hunk 1 above)
Comment (iurisevero) [reply rasa/utils/common.py:315]: Right! I'll add that asap

**Review by wochinge — COMMENTED** [review]
Sweet! Can you please also add a changelog which describes that this fixes the warning for anybody using Rasa Open Source with Python > 3.6?

**Comment by iurisevero** [comment]
> Sweet! Can you please also add a changelog which describes that this fixes the warning for anybody using Rasa Open Source with Python > 3.6?

Sure! Should I name it as ```misc``` or ```ìmprovement``` (or another type)?

File: changelog/8263.improvement.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Updated ```asyncio.Task.all_tasks``` to ```asyncio.all_tasks```, with a fallback for python 3.6, which raises an AtributeError for ```asyncio.all_tasks```.
```
Comment (wochinge) [inline changelog/8263.improvement.md:1]: a single backtick is enough
```suggestion
Updated `asyncio.Task.all_tasks` to `asyncio.all_tasks`, with a fallback for python 3.6, which raises an AtributeError for `syncio.all_tasks`.
```

File: changelog/8263.improvement.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Updated `asyncio.Task.all_tasks` to `asyncio.all_tasks`, with a fallback for python 3.6, which raises an AtributeError for `asyncio.all_tasks`.
```
Comment (wochinge) [inline changelog/8263.improvement.md:1]:
```suggestion
Updated `asyncio.Task.all_tasks` to `asyncio.all_tasks`, with a fallback for python 3.6, which raises an AttributeError for `asyncio.all_tasks`. This removes the deprecation warning for the `Task.all_tasks` usage.
```

**Review by wochinge — APPROVED** [review]
Thanks for your contribution!

**Comment by wochinge** [comment]
Can you please update your branch to latest `main`?

**Comment by iurisevero** [comment]
> Thanks for your contribution!

Thanks for all reviews! xD

**Pull request merged**